src/netex_idf/offers.rs

```diff
@@ -267,13 +283,105 @@ fn enhance_with_object_code(
     enhanced_routes
 }
 
+fn arrival_departure_times(el: &Element) -> Result<(Time, Time)> {
+    fn time(el: &Element, node_name: &str) -> Result<Time> {
+        Ok(el.try_only_child(node_name)?.text().parse()?)
+    }
+    let offset: u32 = match el.try_only_child("DepartureDayOffset")?.text().parse() {
+        Ok(offset) if offset > 0 => offset,
+        Ok(offset) if offset == 0 => 0,
+        _ => 0,
+    };
+    let offset_time = Time::new(24 * offset, 0, 0);
+    let arrival_time = time(el, "ArrivalTime")?;
+    let departure_time = time(el, "DepartureTime")?;
+
+    Ok((arrival_time + offset_time, departure_time + offset_time))
+}
+
+fn boarding_type(el: &Element, node_name: &str) -> u8 {
+    el.only_child(node_name)
+        .map(|node| (node.text() == "false") as u8)
+        .unwrap_or(0)
+}
+
+fn stop_point_idx(
+    sp_in_jp: &Element,
+    stop_points: &CollectionWithId<StopPoint>,
+    map_schedule_stop_point_quay: &HashMap<String, String>,
+) -> Result<Idx<StopPoint>> {
+    sp_in_jp
+        .try_only_child("ScheduledStopPointRef")
+        .and_then(|ssp_ref_el| ssp_ref_el.try_attribute::<String>("ref"))
+        .and_then(|ssp_ref| {
+            map_schedule_stop_point_quay.get(&ssp_ref).ok_or_else(|| {
+                format_err!(
+                    "QuayRef corresponding to ScheduledStopPointRef {} not found",
+                    ssp_ref
+                )
+            })
+        })
+        .and_then(|quay_ref| {
+            stop_points
+                .get_idx(quay_ref)
+                .ok_or_else(|| format_err!("stop point {} not found", quay_ref))
+        })
+}
+
+fn stop_times(
+    service_journey_element: &Element,
+    map_journeypatterns: &HashMap<String, &Element>,
+    stop_points: &CollectionWithId<StopPoint>,
+    map_schedule_stop_point_quay: &HashMap<String, String>,
+) -> Result<Vec<StopTime>> {
+    let sj_el = service_journey_element.only_child("passingTimes");
+    let timetable_passing_times = sj_el.iter().flat_map(|el| el.children());
+
+    let journey_pattern_ref: String = service_journey_element
+        .try_only_child("JourneyPatternRef")?
+        .try_attribute("ref")?;
+    let jp = map_journeypatterns.get(&journey_pattern_ref);
```

Comment:
Same here, you don't really need this intermediate variable.
```rust
let stop_points_in_journey_pattern = map_journeypatterns
    .get(&journey_pattern_ref)
    .into_iter()
    .filter_map(|jp| jp.only_child("pointsInSequence"))
    .flat_map(|p| p.children());
```

Reply:
Same as https://github.com/CanalTP/transit_model/pull/454#discussion_r345116225.